presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcFileWriterFactory.java

```diff
@@ -140,7 +141,7 @@ public OrcWriterStats getStats()
         // an index to rearrange columns in the proper order
         List<String> fileColumnNames = getColumnNames(schema);
         List<Type> fileColumnTypes = getColumnTypes(schema).stream()
-                .map(hiveType -> hiveType.getType(typeManager))
+                .map(hiveType -> hiveType.getType(typeManager, getTimestampPrecision(session).getPrecision()))
```

Comment:
we could have a short-cut `hiveType.getType(typeManager, session)` overload (a potential followup cleanup)